test/e2e/scripts/setup-instance/02-ec2.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+printf '=%.0s' {0..79} ; echo
+set -ex
+set -o pipefail
+
+cd "$(dirname $0)"
+
+export COMMIT_ID=$(git rev-parse --verify HEAD)
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+COMMIT_USER=$(git log -1 --pretty=format:'%an')
+
+# If not using the default value, mind to change the following settings:
+# - AMI
+# - security groups ids
+# - subnet ids
+REGION="${REGION:-us-east-1}"
+
+aws ec2 request-spot-instances \
+    --spot-price "0.015" \
+    --instance-count 1 \
+    --type "one-time" \
+    --valid-until $(($(date +%s) + 3300)) \
+    --launch-specification file://specification.json \
+    --region ${REGION} | tee spot-instance-request.json
+
+SPOT_REQUEST_ID=$(jq -re .SpotInstanceRequests[].SpotInstanceRequestId spot-instance-request.json)
+
+
+while true
+do
+    aws ec2 describe-spot-instance-requests \
```

Comment:
Depending on how the aws cli is installed and the creds are set, would it make sense to make sure we can request/create an instance before looping here ?

Reply:
Why not, how could we do that ?

Reply:
Something along the lines of https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetRole.html ?

Reply:
We are already running with a custom IAM roles, all the permissions are defined before and on a separate repo.

Reply:
yea, I understand. I just figured that if someone modifies the perms, it might affect the e2e